### PR TITLE
ceph: Add RBAC in helm chart for mgr to create service monitor

### PR DESCRIPTION
**Pull request by sathieu**
**Description of your changes:**

Same as #8118, for Helm chart.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by sathieu** [comment]
My understanding was wrong.

I had to set `monitoring.enabled` to `true` to make this work.

Sorry for the noise...

**Comment by BlaineEXE** [comment]
Thank you @sathieu for contributing this. We removed recently this bit of code in the operator helm chart because we don't want to strongly tie installation of Rook to Prometheus. 

In our example manifests, we have `deploy/examples/monitoring/rbac.yaml` which deploys the RBAC needed for Prometheus monitoring for users who manually install, and we have the same optional-ness for users of the `rook-ceph-cluster` Helm chart, which can optionally deploy the monitoring RBAC.

Given that not all users are using the `rook-ceph-cluster` chart, I have proposed a different solution that will optionally install the required helm resources in the `rook-ceph` chart. See here: https://github.com/rook/rook/pull/9388

@sathieu would you care to test that new PR's changes to make sure your workflow is working with it?

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Requesting changes while we evaluate a different solution.

**Comment by sathieu** [comment]
I still have another error:

```
2021-12-10 17:31:23.559425 E | op-mgr: failed to deploy prometheus rule. prometheus rule could not be deployed: failed to create prometheusRules. prometheusrules.monitoring.coreos.com is forbidden: User "system:serviceaccount:rook-ceph:rook-ceph-system" cannot create resource "prometheusrules" in API group "
```

File: deploy/charts/rook-ceph/values.yaml [hunk 1]
```diff
@@ -52,6 +52,10 @@ rbacEnable: true
 ##
 pspEnable: true
 
+monitoring:
+  # enabling will create RBAC rules to allow Operator to create ServiceMonitors
+  enabled: true
+
```
Comment (BlaineEXE) [inline deploy/charts/rook-ceph/values.yaml:58]: Monitoring should not be enabled for all users by default. For our releases that go out to hundreds of users, we'd like this to be false by default, and for those who want monitoring, they enable it via `monitoring.enabled`.

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Also, please change the subject to `helm: ...` for our feature tracking. `ceph: ...` is a bit too generic.

**Comment by BlaineEXE** [comment]
Closing now that https://github.com/rook/rook/pull/9388 is merged.

**Pull request closed without merging**